code/obj/window.dm

```diff
@@ -1044,6 +1044,28 @@
 
 
 // Flockdrone BS goes here - cirr
+/obj/window/auto/feather
+	default_material = "gnesisglass"
+
+/obj/window/auto/feather/New()
+	connects_to += /turf/simulated/wall/auto/feather
+	..()
+	src.AddComponent(/datum/component/flock_protection, FALSE, TRUE, TRUE)
+
+/obj/window/auto/feather/special_desc(dist, mob/user)
+  if(isflock(user))
+    return {"<span class='flocksay'><span class='bold'>###=-</span> Ident confirmed, data packet received.
+    <br><span class='bold'>ID:</span> Fibrewoven Window
+    <br><span class='bold'>System Integrity:</span> [round((src.health/src.health_max)*100)]%
+    <br><span class='bold'>###=-</span></span>"}
+    // todo: damageable walls
+  else
+    return null // give the standard description
```

Comment:
Some indentation stuff that can be fixed here, and comments removed probably

Reply:
willfix

Reply:
Err I mean previous layout was fine, except maybe some lines after return could be indented further, but none of the lines have proper tabbing

Reply:
oh gross, I didn't even notice